packages/typescript-vfs/src/index.ts

```diff
@@ -572,7 +572,6 @@ export function createVirtualCompilerHost(sys: System, compilerOptions: Compiler
       getCanonicalFileName: fileName => fileName,
       getDefaultLibFileName: () => "/" + ts.getDefaultLibFileName(compilerOptions), // '/lib.d.ts',
       // getDefaultLibLocation: () => '/',
-      getDirectories: () => [],
```

Comment:
My recollection is that this is a newish call; I would suggest putting it before the spread so that older runtimes won't crash. The TS website in particular supports very old versions.

Reply:
I tried putting it before, and TypeScript type checking failed because it claimed `getDirectories` was specified twice. I'm all for backward-compatibility; is it OK to use `// @ts-ignore`?

Also, speaking of backward compatibility, as far as I can tell, uses of this object already check for its existence. For example:

https://github.com/microsoft/TypeScript/blob/e70904a24fba0e17c6a8acc9f592c80d41543110/src/compiler/moduleNameResolver.ts#L816

But it might be safer to provide the default.

Reply:
Hm, I guess it's complaining that it always overwrites; it's definitely optional in older versions of TS, I'm just thinking about whether or not this was added to make some other behavior work. I can't imagine why it not being provided _wouldn't_ be equivalent to an empty array, though.

Reply:
Easy way to tell, though; I'll have this PR build a preview and we can play around with the playground there.

Reply:
Thanks!

If we leave it this way, would it be a good idea to make `getDirectories` an *optional* property of `System` and `Host`? It seems to be getting treated that way, and this would enforce checking whether it's defined... (Of course, that's an issue for the main TS repo.)

Reply:
On TS? It's optional in some, required in others. Your linked code is `ModuleResolutionHost` where it's explicitly optional.

Reply:
Ah, and it seems to already be optional in `CompilerHost` (via `ModuleResolutionHost`) which is what this host-building function claims to return anyway. So I think we'd be OK even if the `System` didn't have one as it's supposed to.